scripts/examples/typescript/src/XenAPI.ts

```diff
@@ -0,0 +1,52 @@
+import { JSONRPCClient } from "json-rpc-2.0";
+
+class XapiSession {
+    private client: JSONRPCClient;
+    private session_id: string | undefined;
+
+    constructor(hostUrl: string | undefined) {
+        this.client = new JSONRPCClient(async (request) => {
+            const response = await fetch(`${hostUrl}/jsonrpc`, {
+                method: "POST",
+                headers: { "content-type": "application/json" },
+                body: JSON.stringify(request),
+            });
+            if (response.status === 200) {
+                const json = await response.json();
+                return this.client.receive(json);
+            } else if (request.id !== undefined) {
+                return Promise.reject(new Error(response.statusText));
+            }
+        });
+    }
+
+    async login(method: string, args: any[] = []) {
+        this.session_id = await this.client.request(`session.${method}`, [...args])
+        return this.session_id
+    }
+
+    async xapi_request(method: string, args: any[] = []) {
+        return await this.client.request(`${method}`, [this.session_id, ...args])
```

Comment:
can just pass `method`, you don't need to interpolate the string with <pre>`${method}`</pre>